replicaset.md

```diff
@@ -0,0 +1,190 @@
+are entering an unknown state. 
+
+### How to use a VirtualMachineReplicaSet
+
+The *VirtualMachineReplicaSet* allows to specify a *VirtualMachineTemplate* in
+`spec.template`. It consists of `ObjectMetadata` in `spec.template.metadata`, and
+a `VirtualMachineSpec` in `spec.template.spec`. The specification of the virtual
+machine is equal to the specification of the virtual machine in the
+`VirtualMachine` workload.
+
+`spec.replicas` can be used to specify how much replicas are wanted. If not
+specified, the default value is 1. This value can be updated anytime. The
+controller will react to the changes.
+
+`spec.selector` is used by the controller to keep track of managed virtual
+machines. The selector specified there must be able to match the virtual
+machine labels as specified in `spec.template.metadata.labels`. If the selector
+does not match these labels, or they are empty, the controller will simply do
+nothing except from logging an error. The user is responsible for not creating
+other virtual machines or *VirtualMachineReplicaSets* which conflict with the
+selector and the template labels.
+
+### When to use a VirtualMachineReplicaSet
+
+Whenever one wants many equal Virtual Machines and does not care about keeping
+any kind of state, after they were terminated, the VirtualMachineReplicaSet is
+the right choice.
+
+#### Fast starting ephemeral Virtual Machines
+
+In this use-case one has small, fast booting Virtual Machines, and only minimal or
+no provisioning is done during the boot. With other words, the timespan
+between the creation of a new Virtual Machine and entering the ready state of a
+VirtualMachine is short.
+
+Migrations are not important. Load balancing between nodes can be achieved by
+simply deleting managed VirtualMachines which are running on an overloaded
+Node. The `eviction` of such a VirtualMachine can happen by directly deleting
+the VirtualMachine instance (KubeVirt aware cluster load balancer) or by
+deleting the corresponding Pod where the Virtual Machine runs in (Only
+Kubernetes aware cluster load balancer).
+
+#### Slow starting ephemeral Virtual Machines
+
+In this use-case one has big, slow booting Virtual Machines, and complex or
+resource intensive provisioning is done during the boot. With other words, the
+timespan between the creation of a new Virtual Machine and entering the ready
+state of a VirtualMachine is long.
+
+One still does not care about the state, but since re-provisioning
+VirtualMachines is expensive, Migrations are important. Load balancing between
+nodes can be achieved by migrating VirtualMachines to different nodes. A
+cluster load balancer needs to be aware of KubeVirt and create Migrations,
+instead of `evicting` VirtualMachines by deletion.
+
+> **Note:** The simplest form of having a migratable ephemeral VirtualMachine,
+> will be to use local storage based on `ContainerRegistryDisks` in combination
+> with a file based backing store. However, migratable backing store
```

Comment:
This is a good idea. I forgot it is technically possible to live migrate VMs using registry disks.